src/storage/exec/IndexOutputNode.h

```diff
@@ -0,0 +1,283 @@
+            Row row;
+            auto vId = IndexKeyUtils::getIndexVertexID(planContext_->vIdLen_, val.first);
+            row.emplace_back(Value(std::move(vId)));
+
+            // skip vertexID
+            for (size_t i = 1; i < returnCols.size(); i++) {
+                auto v = IndexKeyUtils::getValueFromIndexKey(planContext_->vIdLen_,
+                                                             vColNum_,
+                                                             val.first,
+                                                             returnCols[i],
+                                                             cols_,
+                                                             false,
+                                                             hasNullableCol_);
+                row.emplace_back(std::move(v));
+            }
+            result_->rows.emplace_back(std::move(row));
+        }
+        return kvstore::ResultCode::SUCCEEDED;
+    }
+
+    kvstore::ResultCode edgeRowsFromData(const std::vector<kvstore::KV>& data) {
+        const auto* schema = type_ == IndexResultType::kEdgeFromDataScan
+                             ? indexEdgeNode_->getSchema()
+                             : indexFilterNode_->getSchema();
+        if (schema == nullptr) {
+            return kvstore::ResultCode::ERR_EDGE_NOT_FOUND;
+        }
+        auto returnCols = result_->colNames;
+        for (const auto& val : data) {
+            Row row;
+            auto src = NebulaKeyUtils::getSrcId(planContext_->vIdLen_, val.first);
+            auto rank = NebulaKeyUtils::getRank(planContext_->vIdLen_, val.first);
+            auto dst = NebulaKeyUtils::getDstId(planContext_->vIdLen_, val.first);
+            row.emplace_back(Value(std::move(src)));
+            row.emplace_back(Value(rank));
+            row.emplace_back(Value(std::move(dst)));
+            auto reader = RowReader::getRowReader(schema, val.second);
+            if (!reader) {
+                VLOG(1) << "Can't get tag reader";
+                return kvstore::ResultCode::ERR_EDGE_NOT_FOUND;
+            }
+            // skip column src_ , edgeType, ranking, dst_
+            for (size_t i = 3; i < returnCols.size(); i++) {
+                auto v = reader->getValueByName(returnCols[i]);
+                row.emplace_back(std::move(v));
+            }
+            result_->rows.emplace_back(std::move(row));
+        }
+        return kvstore::ResultCode::SUCCEEDED;
+    }
+
+    kvstore::ResultCode edgeRowsFromIndex(const std::vector<kvstore::KV>& data) {
+        auto returnCols = result_->colNames;
+        for (const auto& val : data) {
+            Row row;
+            auto src = IndexKeyUtils::getIndexSrcId(planContext_->vIdLen_, val.first);
+            auto rank = IndexKeyUtils::getIndexRank(planContext_->vIdLen_, val.first);
+            auto dst = IndexKeyUtils::getIndexDstId(planContext_->vIdLen_, val.first);
+
+            row.emplace_back(Value(std::move(src)));
```

Comment:
Is it ` row.values.emplace_back(std::move(src))` ？

Reply:
`Row`, a.k.a `List` has a delegate on this method.

Reply:
👍，this method is concise